File: animSmooth/animSmooth/animSmooth.cpp

#include <maya/MSimple.h>
#include <maya/MGlobal.h>
#include <maya/MString.h>
#include <maya/MPxToolCommand.h>
#include <maya/MFnPlugin.h>

class animSmooth : public MPxCommand {
public:
	static void* creator()
	{
		return new animSmooth();
	}
	virtual ~animSmooth() {}
	animSmooth() {}
	virtual MStatus doIt(const MArgList& args);
private:
	void printHello(MString name);
};

MStatus animSmooth::doIt(const MArgList& args) {
	printHello(args.asString(0));
	return MS::kSuccess;
}

void animSmooth::printHello(MString name) {
	MString message = MString("Hello ");
	message += name;
	MGlobal::displayInfo(message);
}
MStatus initializePlugin(MObject obj) {
	MStatus status = MStatus::kSuccess;
	MFnPlugin plugin(obj, "AnimSmooth", "1.0", "Any");
	// Register Command
	MGlobal::executeCommand("global string $gMainWindow;\n setParent $gMainWindow;");
	MGlobal::executeCommand("menu -label \"AnimSmooth\" -parent MayaWindow -tearOff 1 TestMenu;");
	MGlobal::executeCommand(MString("menuItem -label \"Say Hello\" -echoCommand true -command \"python(\\\"execfile(\\\\\\\"" +
		plugin.loadPath() + "/menu.py" + "\\\\\\\")\\\")\" TestMenuItem1;"));
   status = plugin.registerCommand("helloWorld", animSmooth::creator);
   return status;

}

MStatus uninitializePlugin(MObject obj)
{
	MStatus   status = MStatus::kSuccess;
	MFnPlugin plugin(obj);

	MGlobal::executeCommand("deleteUI -m TestMenu");

	status = plugin.deregisterCommand("helloWorld");
	if (!status) {
		status.perror("deregisterCommand");
		return status;
	}

	return status;
}

File: animSmooth.py
import sys
import maya.OpenMaya as OpenMaya
import maya.OpenMayaMPx as OpenMayaMPx
import pymel.core as pm
import maya.cmds as mc
import maya.mel as mel

commandName = 'hello'
menuName = "TestMenu"

class MyCommandClass( OpenMayaMPx.MPxCommand ):
    
    def __init__(self):
        ''' Constructor. '''
        OpenMayaMPx.MPxCommand.__init__(self)
    
    def doIt(self, args):
        ''' Command execution. '''
        
        print 'hello world!'

class GUICommandClass( OpenMayaMPx.MPxCommand):

    def __init__(self):
        ''' Constructor. '''
        OpenMayaMPx.MPxCommand.__init__(self)
    def doIt(self, args):
        window = mc.window( title="Long Name", iconName='Short Name', widthHeight=(200, 55) )
        mc.columnLayout( adjustableColumn=True )
        mc.button( label='Say Hello', command=('import maya.cmds as mc; mc.hello()') )
        mc.showWindow( window )

        
def cmdCreator():
    ''' Create an instance of our command. '''
    return OpenMayaMPx.asMPxPtr( MyCommandClass() )

def guiCreator():
    return OpenMayaMPx.asMPxPtr( GUICommandClass() )

def initializePlugin( mobject ):
    ''' Initialize the plug-in when Maya loads it. '''
    mplugin = OpenMayaMPx.MFnPlugin( mobject )
    gMainWindow = mel.eval('$tmpVar=$gMainWindow')
    mc.setParent ( gMainWindow )
    if pm.menu(menuName, exists=True):
        print("Menu found")
    else:
        mc.menu( menuName, parent=gMainWindow, label=menuName)
        mc.menuItem( menuName+"Item1", label="Hello", command="import maya.cmds as mc; mc.hello()")
        mc.menuItem( menuName+"Item2", label="HelloGUI", command="import maya.cmds as mc; mc.hellogui()")
    try:
        mplugin.registerCommand( commandName, cmdCreator )
        mplugin.registerCommand( "hellogui", guiCreator)
    except:
        sys.stderr.write( 'Failed to register command: ' + commandName )

def uninitializePlugin( mobject ):
    ''' Uninitialize the plug-in when Maya un-loads it. '''
    pm.deleteUI(menuName)
    mplugin = OpenMayaMPx.MFnPlugin( mobject )
    try:
        mplugin.deregisterCommand( commandName )
        mplugin.deregisterCommand( 'hellogui' )
    except:
        sys.stderr.write( 'Failed to unregister command: ' + commandName )
